sdk/helper/clientcountutil/clientcountutil.go

```diff
@@ -0,0 +1,385 @@
+// Package clientcountutil provides a library to generate activity log data for
+// testing.
+package clientcountutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/clientcountutil/generation"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// ActivityLogDataGenerator holds an ActivityLogMockInput. Users can create the
+// generator with NewActivityLogData(), add content to the generator using
+// the fluent API methods, and generate and write the JSON representation of the
+// input to the Vault API.
+type ActivityLogDataGenerator struct {
+	data            *generation.ActivityLogMockInput
+	addingToMonth   *generation.Data
+	addingToSegment *generation.Segment
+	client          *api.Client
+}
+
+// NewActivityLogData creates a new instance of an activity log data generator
+// The type returned by this function cannot be called concurrently
+func NewActivityLogData(client *api.Client) *ActivityLogDataGenerator {
+	return &ActivityLogDataGenerator{
+		client: client,
+		data:   new(generation.ActivityLogMockInput),
+	}
+}
+
+// NewCurrentMonthData opens a new month of data for the current month. All
+// clients will continue to be added to this month until a new month is created
+// with NewPreviousMonthData.
+func (d *ActivityLogDataGenerator) NewCurrentMonthData() *ActivityLogDataGenerator {
+	d.newMonth(&generation.Data{Month: &generation.Data_CurrentMonth{CurrentMonth: true}})
+	return d
+}
+
+// NewPreviousMonthData opens a new month of data, where the clients will be
+// recorded as having been seen monthsAgo months ago. All clients will continue
+// to be added to this month until a new month is created with
+// NewPreviousMonthData or NewCurrentMonthData.
+func (d *ActivityLogDataGenerator) NewPreviousMonthData(monthsAgo int) *ActivityLogDataGenerator {
+	return d.newMonth(&generation.Data{Month: &generation.Data_MonthsAgo{MonthsAgo: int32(monthsAgo)}})
+}
+
+func (d *ActivityLogDataGenerator) newMonth(newMonth *generation.Data) *ActivityLogDataGenerator {
+	d.data.Data = append(d.data.Data, newMonth)
+	d.addingToMonth = newMonth
+	d.addingToSegment = nil
```

Comment:
it might have data, but that's ok. `d.addingToSegment` is for convenience, so that we know which segment any data should be added to. Because we're starting a new month, we want to stop working with this segment, and create a new one. The data itself still lives in d.data.Data

Reply:
Thanks for the context!